File: src/types/task.ts
/**
 * 任务类型定义
 */
export interface Task {
  id: string;
  title: string;
  description?: string;
  completed: boolean;
  spicyLevel: number; // 1-5，表示任务难度
  category?: string; // 任务分类
  subtasks: Task[];
  createdAt: Date;
  updatedAt: Date;
}

/**
 * AI 拆解请求参数
 */
export interface BreakdownRequest {
  task: string;
  goal?: string;
  spicyLevel: number;
  existingTasks?: string[]; // 已有任务标题列表，用于避免重复
}

/**
 * AI 拆解响应
 */
export interface BreakdownResponse {
  subtasks: Omit<Task, 'id' | 'subtasks' | 'createdAt' | 'updatedAt'>[];
}

File: new-project/backend/app/models/task.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, DateTime, Boolean, Integer, Text, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.db.database import Base


class Task(Base):
    __tablename__ = "tasks"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    parent_id = Column(UUID(as_uuid=True), ForeignKey("tasks.id", ondelete="CASCADE"), nullable=True)
    title = Column(String(500), nullable=False)
    description = Column(Text, default="")
    completed = Column(Boolean, default=False)
    spicy_level = Column(Integer, default=3)
    category = Column(String(100), default="")
    sort_order = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    user = relationship("User", back_populates="tasks")
    parent = relationship("Task", remote_side=[id], back_populates="children")
    children = relationship("Task", back_populates="parent", cascade="all, delete-orphan")